farmdata2_modules/fd2_tabs/resources/RegexInputComponent.spec.comp.js

```diff
@@ -0,0 +1,228 @@
+import { shallowMount } from '@vue/test-utils'
+
+var RegexInputComp = require("./RegexInputComponent.js")
+var RegexInputComponent = RegexInputComp.RegexInputComponent
+
+describe('RegexInput Component', () => {
+
+        context('normal feature tests', () => {
+            beforeEach(() => {
+                mount(RegexInputComponent, {
+                    propsData: {
+                        regExp: null,
+                    }
+                })
+            })
+
+            it('exists', () => {
+                cy.get('[data-cy=regex-input]').should('exist')
+            })
+
+            it('can be typed in', () => {
+                cy.get('[data-cy=text-input]')
+                    .clear()
+                    .type('Hello World!')
+                    .should('have.value', 'Hello World!')
+        })
+
+            it('emits the new isMatch value after blur', () => {
+                const spy = cy.spy()
+                Cypress.vue.$on('is-match-changed', spy)
+                cy.get('[data-cy=text-input]')
+                    .type('Hello world!')
+                    .blur()
+                    .then(() => {
+                        expect(spy).to.be.called
+                    })
+    })
+})
+
+    context('testing disabled option', () => {
+        let comp;
+        beforeEach(() => {
+            //ShallowMount to test disabled option
+            comp = shallowMount(RegexInputComponent, { 
+                propsData: {
+                    regExp: null,
+                    defaultVal: null,
+                    disabled: true,
+                },
+            })
+        })
+    
+    it('checking if the input box is disabled', () => {
+        mount(RegexInputComponent, {
+            propsData: {
+                regExp: null,
+                defaultVal: null,
+                disabled: true,
+            }
+        })
+        cy.get('[data-cy=text-input]')
+        .should('be.disabled')
+    })
+    it('checking disabled when prop changes', () => {
+        expect(comp.vm.isDisabled).to.equal(true)
+        cy.wrap(comp.setProps({ disabled: false })).then(() => {
+            expect(comp.vm.isDisabled).to.equal(false)
+        })
+    })
+})
+
+        context('watch prop change tests', () => {
+            let comp;
+            beforeEach(() => {
+                comp = shallowMount(RegexInputComponent, {
+                    propsData: {
+                        regExp: null,
+                        defaultVal: null,
+                        setType: 'text',
+                        setColor: 'pink',
+                        setHeight: '25px',
+                        setWidth: '4em',
+                        setMin: null,
+                        setMax: null,
+                        setStep: null,
+                        setTitle: null,
+                        disabled: null
+
+                    },
+                })
+            })
+
+            it('change regExp prop', () => {
```

Comment:
Do all of these prop changes work?  I don't see watch statements for all of them in the component code.  Isn't it necessary to have a watch in the component for the change in prop to be detected in the component?  If these tests pass it seems not, maybe we need to investigate this some more?